platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java

```diff
@@ -105,17 +116,26 @@ public boolean isIllegalDirectSchedulerUsagePresent() {
      * Wire the components together.
      */
     private void wire() {
-        internalEventValidatorScheduler.getEventOutput().solderTo(eventDeduplicatorScheduler.getEventInput());
-        eventDeduplicatorScheduler.getEventOutput().solderTo(eventSignatureValidatorScheduler.getEventInput());
-        eventSignatureValidatorScheduler.getEventOutput().solderTo(orphanBufferScheduler.getEventInput());
-        orphanBufferScheduler.getEventOutput().solderTo(inOrderLinkerScheduler.getEventInput());
-        inOrderLinkerScheduler.getEventOutput().solderTo(linkedEventIntakeScheduler.getEventInput());
+        internalEventValidatorWiring.eventOutput().solderTo(eventDeduplicatorWiring.eventInput());
+        eventDeduplicatorWiring.eventOutput().solderTo(eventSignatureValidatorWiring.eventInput());
+        eventSignatureValidatorWiring.eventOutput().solderTo(orphanBufferWiring.eventInput());
+        orphanBufferWiring.eventOutput().solderTo(inOrderLinkerWiring.eventInput());
+        inOrderLinkerWiring.eventOutput().solderTo(linkedEventIntakeWiring.eventInput());
+
+        final OutputWire<Long> minimumGenerationNonAncientOutput =
+                linkedEventIntakeWiring.minimumGenerationNonAncientOutput();
+
+        minimumGenerationNonAncientOutput.solderTo(eventDeduplicatorWiring.minimumGenerationNonAncientInput(), INJECT);
+        minimumGenerationNonAncientOutput.solderTo(
+                eventSignatureValidatorWiring.minimumGenerationNonAncientInput(), INJECT);
+        minimumGenerationNonAncientOutput.solderTo(orphanBufferWiring.minimumGenerationNonAncientInput(), INJECT);
+        minimumGenerationNonAncientOutput.solderTo(inOrderLinkerWiring.minimumGenerationNonAncientInput(), INJECT);
```

Comment:
Optional suggestion.

As we add more and more things that need to be soldered, this function is going to get large. I wonder if it would make sense to split certain related wiring things into helper functions. For example, we could create a function `solderMinimumGenerationNonAncient()` that encapsulates these lines.